src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -340,22 +347,25 @@ export class IndexPatternsService {
    * @returns IndexPatternSpec
    */
 
-  savedObjectToSpec = (savedObject: SavedObject<IndexPatternAttributes>): IndexPatternSpec => {
+  savedObjectToSpec = async (
+    savedObject: SavedObject<IndexPatternAttributes>
+  ): Promise<IndexPatternSpec> => {
     const {
       id,
       version,
       attributes: {
-        title,
-        timeFieldName,
-        intervalName,
-        fields,
-        sourceFilters,
+        allowNoIndex,
```

Comment:
Apologies for the extra line changes. It is so much easier to see where updates have been made to objects/types when their properties are alphabetized, especially with bigger objects. I did this in a few places throughout this PR